# Prenatal Alcohol Exposure: Profiling Developmental DNA Methylation Patterns in Central and Peripheral Tissues

## A. Manuscript
  • Citation: Lussier AA, Bodnar TS, Mingay M, Morin AM, Hirst M, Kobor MS, Weinberg J. Prenatal Alcohol Exposure: Profiling Developmental DNA Methylation Patterns in Central and Peripheral Tissues. Front Gen. 2018.
  
  • doi: 10.3389/fgene.2018.00610
  
  • GEO accession number: [GSE121582](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE121582)
  
  • Source: https://www.frontiersin.org/articles/10.3389/fgene.2018.00610/full

  • Supplementary materials

## B. Scripts
  • Preprocessing pipeline
    
  • Analysis - Developmental
    
  • Analysis - Tissue-concordant
    
  • Analysis - Age-related
    
  • Annotation 
    
  • Other

## C. Project summary
**Background**: Prenatal alcohol exposure (PAE) can alter the development of neurobiological systems, leading to lasting neuroendocrine, neuroimmune, and neurobehavioral deficits. Although the etiology of this reprogramming remains unknown, emerging evidence suggests DNA methylation as a potential mediator and biomarker for the effects of PAE due to its responsiveness to environmental cues and relative stability over time. Here, we utilized a rat model of PAE to examine the DNA methylation profiles of rat hypothalami and leukocytes at four time points during early development to assess the genome-wide impact of PAE on the epigenome and identify potential biomarkers of PAE. 

**Methods**: Our model of PAE resulted in blood alcohol levels of ~80–150 mg/dl throughout the equivalent of the first two trimesters of human pregnancy. Hypothalami were analyzed on postnatal days (P) 1, 8, 15, 22 and leukocytes at P22 to compare central and peripheral markers. Genome-wide DNA methylation analysis was performed by methylated DNA immunoprecipitation followed by next-generation sequencing.

**Results**: PAE resulted in lasting changes to DNA methylation profiles across all four ages, with 118 differentially methylated regions (DMRs) displaying persistent alterations across the developmental period at a false-discovery rate (FDR) < 0.05. In addition, 299 DMRs showed the same direction of change in the hypothalamus and leukocytes of P22 pups at an FDR < 0.05, with some genes overlapping with the developmental profile findings. The majority of these DMRs were located in intergenic regions, which contained several computationally-predicted transcription factor binding sites. Differentially methylated genes were generally involved in immune function, epigenetic remodeling, metabolism, and hormonal signaling, as determined by gene ontology analyses.

**Conclusions**: Persistent DNA methylation changes in the hypothalamus may be associated with the long-term physiological and neurobehavioral alterations in observed in PAE. Furthermore, correlations between epigenetic alterations in peripheral tissues and those in the brain will provide a foundation for the development of biomarkers of fetal alcohol spectrum disorder (FASD). Finally, findings from studies of PAE provide important insight into the etiology of neurodevelopmental and mental health disorders, as they share numerous phenotypes and comorbidities.
